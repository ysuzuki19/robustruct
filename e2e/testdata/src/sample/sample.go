package sample

type Sample struct {
	A  int
	B  string
	AA bool
}

func Sample1() {
	s := Sample{ // want "fields 'AA' are not initialized"
		A: 1,
		B: "hello",
	}
	_ = s
}

func Sample2() {
	s := Sample{ // want "all fields of the struct must be sorted by defined order"
		B:  "hello",
		A:  1,
		AA: false,
	}
	_ = s
}

func Sample3() {
	// ignore:robustruct
	s := Sample{
		B: "hello",
		A: 1,
	}
	_ = s
}

func Sample4() {
	s := Sample{
		A: 1,
		B: "hello",
	} // ignore:fields_require
	_ = s
}
