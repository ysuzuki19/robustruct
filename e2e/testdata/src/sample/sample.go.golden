package sample

type Sample struct {
	A  int
	B  string
	AA bool
}

func Sample1() {
	s := Sample{ // want "fields 'AA' are not initialized"
		A:  1,
		B:  "hello",
		AA: false,
	}
	_ = s
}

func Sample2() {
	s := Sample{
		A:  1,
		B:  "hello",
		AA: false,
	}
	_ = s
}

func Sample3() {
	// ignore:robustruct
	s := Sample{
		B: "hello",
		A: 1,
	}
	_ = s
}

func Sample4() {
	s := Sample{
		A: 1,
		B: "hello",
	} // ignore:fields_require
	_ = s
}
